android/src/main/java/com/pylon/RNSpokestack/RNSpokestackModule.java

```diff
@@ -17,101 +19,138 @@
+            for (String k : map.keySet())
+                ttsBuilder.setProperty(k, map.get(k));
+
+            ttsBuilder.addTTSListener(this);
+            tts = ttsBuilder
+                    .setTTSServiceClass(map.get("ttsServiceClass").toString())
+                    .setAndroidContext(reactContext.getApplicationContext())
+                    .build();
+        }
+    }
+
+    @ReactMethod
+    public void start() throws Exception {
+        pipeline.start();
+        onEvent("start", pipeline.getContext().isActive());
+    }
 
-  @Override
-  public void onCatalystInstanceDestroy() {
-    super.onCatalystInstanceDestroy();
-    if (pipeline != null && pipeline.isRunning()) {
-      pipeline.stop();
+    @ReactMethod
+    public void stop () {
+        pipeline.stop();
+        onEvent("stop", pipeline.getContext().isActive());
     }
-  }
 
-  private void sendEvent(String eventName, @Nullable WritableMap params) {
-    if (this.reactContext.hasActiveCatalystInstance()) {
-      this.reactContext
-              .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
-              .emit(eventName, params);
+    @ReactMethod
+    public void activate () {
+        pipeline.getContext().setActive(true);
+        pipeline.getContext().dispatch(SpeechContext.Event.ACTIVATE);
     }
-  }
 
-  @ReactMethod
-  public void initialize(ReadableMap config) {
-    assert config.hasKey("input") : "'input' key is required in config";
-    assert config.hasKey("stages") : "'stages' key is required in config";
+    @ReactMethod
+    public void deactivate () {
+        pipeline.getContext().setActive(false);
+        pipeline.getContext().dispatch(SpeechContext.Event.DEACTIVATE);
+    }
 
-    final SpeechPipeline.Builder builder = new SpeechPipeline.Builder();
-    builder.setInputClass(config.getString("input"));
+    @ReactMethod
+    public void synthesize (ReadableMap ttsInput) {
+        SynthesisRequest req = new SynthesisRequest.Builder(ttsInput.getString("input"))
+                .withMode(SynthesisRequest.Mode.values()[ttsInput.getInt("format")])
+                .withVoice( ttsInput.getString("voice"))
+                .build();
+        tts.synthesize(req);
+    }
+
+    public void onEvent(SpeechContext.Event event, SpeechContext context) {
+        WritableMap react_event = Arguments.createMap();
+        react_event.putString("event", event.name());
+        react_event.putString("transcript", context.getTranscript());
+        react_event.putString("message", context.getMessage());
+        react_event.putString("error", Log.getStackTraceString(context.getError()));
+        react_event.putBoolean("isActive", context.isActive());
+        sendEvent("onSpeechEvent", react_event);
+    }
 
-    for (Object stage : config.getArray("stages").toArrayList()) {
-      builder.addStageClass(stage.toString());
+    public void onEvent(String event, Boolean active) {
```

Comment:
Why are init, start, and stop dispatched as separate events? They aren't events in the Android version, and they don't mean much—init doesn't do anything but construct the object (stages aren't built until start is called), and start and stop have to be called explicitly, so presumably you know you've called them.

Reply:
These events are for cross-platform consistency in react-native.